python_ta/transforms/type_inference_visitor.py

```diff
@@ -124,7 +124,10 @@ def set_binop_type_constraints(node):
 
 
 def set_unaryop_type_constraints(node):
-    node.type_constraints = node.operand.type_constraints
+    if node.op == 'not':
+        node.type_constraints = TypeInfo(bool)
+    else:
+        node.type_constraints = node.operand.type_constraints
```

Comment:
You can do better than this. Look up the operation and the corresponding method, and the `unify_call` it.